python/tk_multi_launchapp/base_launcher.py

```diff
@@ -270,18 +273,26 @@ def _launch_callback(self, menu_name, app_engine, app_path, app_args, version=No
             entity_type = self._tk_app.context.task["type"]
             entity_id = self._tk_app.context.task["id"]
 
-        # Now do the folder creation. If there is a specific defer keyword, this takes
-        # precedence. Otherwise, use the engine name for the DCC application by default.
-        defer_keyword = self._tk_app.get_setting("defer_keyword") or app_engine
-        try:
-            self._tk_app.log_info("Creating folders for %s %s. Defer keyword: '%s'" %
-                (entity_type, entity_id, defer_keyword)
-            )
-            self._tk_app.sgtk.create_filesystem_structure(
-                entity_type, entity_id, engine=defer_keyword
+        if len(self._tk_app.sgtk.roots) == 0:
```

Comment:
this is the only actual part that's changed. The rest was just cleanup of some older non-pep8 code.